src/libraries/System.Memory/tests/Span/IndexOfAny.AlgorithmicComplexity.cs

```diff
@@ -0,0 +1,149 @@
+            // scratch buffer so that it extends into the poison page. If the runtime skips past the
+            // first occurrence of the needle and attempts to read all the way to the end of the span,
+            // this will manifest as an AV within this unit test.
+
+            using BoundedMemory<T> boundedMem = BoundedMemory.Allocate<T>(4096, PoisonPagePlacement.After);
+            Span<T> span = boundedMem.Span;
+            span.Clear();
+
+            span = MemoryMarshal.CreateSpan(ref MemoryMarshal.GetReference(span), span.Length + 4096);
+
+            for (int i = 0; i < needle.Length; i++)
+            {
+                span[1024] = needle[i];
+                Assert.Equal(1024, MemoryExtensions.IndexOfAny(span, needle));
+                Assert.Equal(1024, MemoryExtensions.IndexOfAny((ReadOnlySpan<T>)span, needle));
+            }
+        }
+
+        private static void RunLastIndexOfAnyAlgComplexityTest<T>(T[] needle) where T : unmanaged, IEquatable<T>
+        {
+            // Similar to RunIndexOfAnyAlgComplexityTest (see comments there), but we run backward
+            // since we're testing LastIndexOfAny.
+
+            using BoundedMemory<T> boundedMem = BoundedMemory.Allocate<T>(4096, PoisonPagePlacement.Before);
+            Span<T> span = boundedMem.Span;
+            span.Clear();
+
+            span = MemoryMarshal.CreateSpan(ref Unsafe.Subtract(ref MemoryMarshal.GetReference(span), 4096), span.Length + 4096);
+
+            for (int i = 0; i < needle.Length; i++)
+            {
+                span[^1024] = needle[i];
+                Assert.Equal(span.Length - 1024, MemoryExtensions.LastIndexOfAny(span, needle));
+                Assert.Equal(span.Length - 1024, MemoryExtensions.LastIndexOfAny((ReadOnlySpan<T>)span, needle));
+            }
+        }
+
+        // returns [ 'a', 'b', 'c', ... ], or the equivalent in bytes, ints, etc.
+        private static IEnumerable<T> GetIndexOfAnyNeedlesForAlgComplexityTest<T>() where T : unmanaged
+        {
+            for (int i = 0; i < 26; i++)
+            {
+                yield return (T)Convert.ChangeType('a' + i, typeof(T), CultureInfo.InvariantCulture);
+            }
+        }
+
+#pragma warning disable CS0659 // Type overrides Object.Equals(object o) but does not override Object.GetHashCode()
+        private sealed class CustomEquatableType<T> : IEquatable<CustomEquatableType<T>> where T : IEquatable<T>
+#pragma warning restore CS0659 // Type overrides Object.Equals(object o) but does not override Object.GetHashCode()
+        {
+            private readonly T _value;
+            private readonly bool _isPoison;
+
+            public CustomEquatableType(T value, bool isPoison = false)
+            {
+                _value = value;
+                _isPoison = isPoison;
+            }
+
+            public override bool Equals(object? obj) => base.Equals(obj);
```

Comment:
Will fix this call site. It doesn't change the unit test since the unit test never exercises this overload of _Equals_.